gui/packages/desktop/src/renderer/app.js

```diff
@@ -360,18 +360,78 @@ export default class AppRenderer {
     actions.settings.updateAutoConnect(autoConnect);
   }
 
+  async _getAppComponentsVersions() {
+    const daemonVersion = await this._daemonRpc.getCurrentVersion();
+    const guiVersion = remote.app.getVersion().replace('.0', '');
+    return {
+      daemon: daemonVersion,
+      gui: guiVersion,
+      isConsistent: daemonVersion === guiVersion,
+    };
+  }
+
   async _fetchCurrentVersion() {
     const actions = this._reduxActions;
-    const versionFromDaemon = await this._daemonRpc.getCurrentVersion();
-    const versionFromGui = remote.app.getVersion().replace('.0', '');
+    const versions = await this._getAppComponentsVersions();
+
+    // notify user about inconsistent version
+    if (process.env.NODE_ENV !== 'development' && !versions.isConsistent) {
+      this._notificationController.notifyInconsistentVersion();
```

Comment:
We can notify the user ASAP about any inconsistencies, so that's why this notification is posted from here.